exporters/otlp/trace/src/main/java/io/opentelemetry/exporter/otlp/trace/OtlpGrpcSpanExporter.java

```diff
@@ -179,6 +179,9 @@ public static OtlpGrpcSpanExporter getDefault() {
    */
   @Override
   public CompletableResultCode shutdown() {
+    if (managedChannel.isShutdown()) {
+      return CompletableResultCode.ofSuccess();
+    }
```

Comment:
There's definitely still a race condition in here, where the shutdown can complete in between this check and the addition of the 2nd notification listener. I'm not sure if there's a solution to that or not.

Reply:
can you add the listener first and do it like this?
```java
final CompletableResultCode result = new CompletableResultCode();
managedChannel.notifyWhenStateChanged(ConnectivityState.SHUTDOWN, result::succeed);
if (managedChannel.isShutdown()) {
  return result.succeed();
}
managedChannel.shutdown();
```

Reply:
That does seem tighter. Depending on the grpc internals, there might still be a race, but this change will definitely make it less likely. Thanks for the suggestion!